src/TypeScript/EslintBridgeClient/EslintBridgeHttpWrapper.cs

```diff
@@ -31,44 +31,55 @@ namespace SonarLint.VisualStudio.TypeScript.EslintBridgeClient
 {
     internal interface IEslintBridgeHttpWrapper : IDisposable
     {
-        Task<string> PostAsync(Uri serverEndpoint, object request, CancellationToken cancellationToken);
+        Task<string> PostAsync(string serverEndpoint, object request, CancellationToken cancellationToken);
     }
 
     internal sealed class EslintBridgeHttpWrapper : IEslintBridgeHttpWrapper
     {
         private readonly ILogger logger;
         private readonly HttpClient httpClient;
+        private readonly IEslintBridgeProcess eslintBridgeProcess;
 
-        public EslintBridgeHttpWrapper(ILogger logger)
-            : this(new HttpClientHandler(), logger)
+        public EslintBridgeHttpWrapper(IEslintBridgeProcessFactory eslintBridgeProcessFactory, ILogger logger)
+            : this(eslintBridgeProcessFactory, new HttpClientHandler(), logger)
         {
         }
 
-        internal EslintBridgeHttpWrapper(HttpMessageHandler httpHandler, ILogger logger)
+        internal EslintBridgeHttpWrapper(IEslintBridgeProcessFactory eslintBridgeProcessFactory,
+            HttpMessageHandler httpHandler,
+            ILogger logger)
         {
             this.logger = logger;
             httpClient = new HttpClient(httpHandler);
+
+            eslintBridgeProcess = eslintBridgeProcessFactory.Create();
         }
 
-        public async Task<string> PostAsync(Uri serverEndpoint, object request, CancellationToken cancellationToken)
+        public async Task<string> PostAsync(string serverEndpoint, object request, CancellationToken cancellationToken)
         {
+            var port = await eslintBridgeProcess.Start();
```

Comment:
As we discussed offline, I believe the init-linter logic should be here as well.
At the moment, if the process crashes, there is no place that does the previous logic  of `if port has changed then call init-linter`. So what will happen is:
* we will spin a new process 
* call analyze-js on it and fail because init-linter was not called
* the exception will be caught in JavascriptAnalyzer that will mark "shouldInitLinter=true"
* the next analysis will succeed

I think it's not ideal, so I do think the init-linter logic should be in the same place that is responsible for spinning up a new process.

Reply:
So far, this http wrapper class knows nothing about any of the available eslintbridge APIs, so it seems weird that it should have special logic to know about one of them i.e. `init-linter`.

Also, `init-linter` isn't required if the caller just wants to call `tsconfig-files` to resolve the config-to-source mappings e.g. if we spun up a separate process just to do that as we discussed this morning. Calling `init-linter` wouldn't be a major overhead in that case, but it's a design smell that the division of responsibilities of this class and its consumers isn't correct.